File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
/*
* 자바의 자료구조에 조금 약한 것 같아서 쉬운 문제들 위주로 풀어 보는 중
* stream 사용법도 익혀야 함!
*/

import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        List<Integer> ans = new ArrayList<>();
        ans.add(arr[0]);
        for(int i = 1; i < arr.length; i++) {
            if(arr[i] != arr[i-1]) ans.add(arr[i]);
        }
        
        int[] answer = ans.stream().mapToInt(Integer::intValue).toArray();
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <queue>
using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    int tmp = arr[0];
    answer.push_back(tmp);
    for (int i = 1; i < arr.size(); i++) {
        if(tmp == arr[i]) continue;
        else {
            answer.push_back(arr[i]);
            tmp = arr[i];
        } 
    }
    
    return answer;
}